test/torch/federated/test_dataset.py

```diff
@@ -20,13 +20,10 @@ def test_federated_dataset(workers):
     data, _ = grid.search("data")
     target, _ = grid.search("target")
 
-    fd = sy.FederatedDataset(data, target, batch_size=32, num_iterators=1)
+    dataset = sy.FederatedDataset(data, target)
+    train_loader = sy.FederatedDataLoader(dataset, batch_size=4, shuffle=False, drop_last=False)
 
-    for iter in range(5):
-        fd.reset()
-        i = 0
-        loss_accum = 0
-        while fd.keep_going():
-            i += 1
-
-            data, target = fd.step()
+    epochs = 2
+    for epoch in range(1, epochs + 1):
+        for batch_idx, (data, target) in enumerate(train_loader):
+            pass
```

Comment:
what is this testing? This seems like it purely makes pytest think that that code was run while testing nothing

Reply:
It is testing a no failing - we're a bit short of time so I didn't add exptensive tests

Reply:
Is there a way that we can test this for real?

Reply:
Good question - strictly speaking it's job is to just feed data. I suppose one could check:
- to make sure it feeds all the data once per iteration
- to make sure it doesn't feed remote data from different workers.

but neither are worth blocking this at the moment.

Reply:
Valid. We should open an issue for it

Reply:
👍